src/SparkTestBase.sol

```diff
@@ -40,22 +45,134 @@ interface IExecutable {
     function execute() external;
 }
 
-abstract contract SparkTestBase is Test {
-    address internal executor;
-    address internal payload;
+abstract contract SpellRunner is Test {
+    using DomainHelpers for Domain;
+    using DomainHelpers for StdChains.Chain;
+
+    mapping(ChainId chainId => IExecutor executor) internal executors;
```

Comment:
I also thought of making a big struct with all chain-specific data, which would have the benefit of chain-dependent configuration being a mapping->struct access instead of ad-hoc `if`s (see `_getPoolAddressesProviderRegistry`), opinions?

Reply:
I'm in favor of using a big struct here with potentially some values set to zero. For example, Base doesn't have pool address registry, so that can be address 0 and we error if someone tries to load SparkLend on a chain where that doesn't exist.

Reply:
implemented in 5890d9e